src/io/metadata.cpp

```diff
@@ -214,14 +216,17 @@ void Metadata::CheckOrPartition(data_size_t num_all_data, const std::vector<data
 
 
 void Metadata::SetInitScore(const float* init_score, data_size_t len) {
-  if (num_data_ != len) {
+  if (num_data_ == 0 || len % num_data_ != 0) {
```

Comment:
@chivee I checked the size of init scores in score_updater to avoid this bug. It doesn't matter now, because I use another logic in my last commit.